### PR TITLE
Add protocol/protocol-name docs and test

### LINKED ISSUE #121: Missing documentation/unit tests about protocol options
Found out by looking at source code there are missing documentation about the `protocols` option which gets used in [electron-packager/mac.js](https://github.com/maxogden/electron-packager/blob/303de520cef5a46d61a7a8646e9d83aa079af20a/mac.js) for building electron apps that register custom url scheme on os. Example when opening `myapp://blip-blop` in browser it passes `myapp://blip-blop` string as an argument to [`open-url`](https://github.com/atom/electron/blob/master/docs/api/app.md#event-open-url) app module event.

### PR DESCRIPTION
**Pull request by kevinsawicki**
Adds documentation and an initial test for the `protocol`/`protocol-name` option.

Closes #121

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -276,6 +276,23 @@ If present, signs OS X target apps when the host platform is OS X and XCode is i
...
 
+##### `protocol`
+
+*String*
+
+The URL protocol scheme to associate the app with. Specifying `myapp` would
+cause URLs such as `myapp://path` to be opened with the app. This option can be
+specified multiple times to register multiple protocol schemes with the app.
+Maps to the `CFBundleURLSchemes` metadata property. This argument requires
+a corresponding `--protocol-name` argument.
```
Comment (malept) [inline docs/api.md]: This is currently written more CLI-centric than API-centric (although it's a little too verbose for `usage.txt`). `s/argument/option/g` and remove the flag prefix from `protocol-name`.

File: docs/api.md [hunk 2]
```diff
@@ -276,6 +276,23 @@ If present, signs OS X target apps when the host platform is OS X and XCode is i
 - `entitlements` (*String*): The path to the 'parent' entitlements.
 - `entitlements-inherit` (*String*): The path to the 'child' entitlements.
 
+##### `protocol`
+
+*String*
```
Comment (malept) [inline docs/api.md]: Should be

> *Array* of *Strings*

**Comment by malept** [comment]
Needs some docs in `usage.txt`.

**Comment by kevinsawicki** [comment]
@malept I've updated `usage.txt`, thanks for the review 👍

File: docs/api.md [hunk 3]
```diff
@@ -288,6 +288,23 @@ If present, signs OS X target apps when the host platform is OS X and XCode is i
 - `entitlements` (*String*): The path to the 'parent' entitlements.
 - `entitlements-inherit` (*String*): The path to the 'child' entitlements.
 
+##### `protocol`
+
+*Array* of *String*s
+
+The URL protocol scheme to associate the app with. Specifying `myapp` would
```
Comment (malept) [inline docs/api.md]: > The URL protocol scheme

scheme(s)

File: docs/api.md [hunk 4]
```diff
@@ -288,6 +288,23 @@ If present, signs OS X target apps when the host platform is OS X and XCode is i
...
+cause URLs such as `myapp://path` to be opened with the app. This option can be
+specified multiple times to register multiple protocol schemes with the app.
```
Comment (malept) [inline docs/api.md]: > This option can be specified multiple times to register multiple protocol schemes with the app.

:scissors:

File: docs/api.md [hunk 5]
```diff
@@ -292,17 +292,16 @@ If present, signs OS X target apps when the host platform is OS X and XCode is i
 
 *Array* of *String*s
 
-The URL protocol scheme to associate the app with. Specifying `myapp` would
-cause URLs such as `myapp://path` to be opened with the app. This option can be
-specified multiple times to register multiple protocol schemes with the app.
-Maps to the `CFBundleURLSchemes` metadata property. This option requires
-a corresponding `protocol-name` option to be specified.
+The URL protocol scheme(s) to associate the app with. Specifying `myapp` would
```
Comment (malept) [inline docs/api.md]: Last thing, I swear:

> For example, specifying `myapp` [...]

**Pull request merged**